sdks/python/apache_beam/io/fileio.py

```diff
@@ -95,18 +87,15 @@ def is_valid_compression_type(cls, compression_type):
   @classmethod
   def mime_type(cls, compression_type, default='application/octet-stream'):
     mime_types_by_compression_type = {
-        cls.GZIP: 'application/x-gzip',
-        cls.ZLIB: 'application/octet-stream',
         cls.BZIP2: 'application/x-bz2',
+        cls.GZIP: 'application/x-gzip',
     }
     return mime_types_by_compression_type.get(compression_type, default)
 
   @classmethod
   def detect_compression_type(cls, file_path):
     """Returns the compression type of a file (based on its suffix)"""
-    compression_types_by_suffix = {'.gz': cls.GZIP,
-                                   '.z': cls.ZLIB,
```

Comment:
Is the removal of zlib intentional? I don't know how many users there are, but it's not backwards compatible.